app/helpers/record_helper.rb

```diff
@@ -4,6 +4,31 @@ def is_number?(object)
     true if Float(object) rescue false
   end
 
+  def has_associated_granule?(record)
+    record.associated_granule_value.present? && is_number?(record.associated_granule_value)
+  end
+
+  def granule_valid?(granule_record, collection_state, associating_flag)
```

Comment:
a little less sure about this, but I would also ask if it makes sense to make this an instance method for either granules or records

Reply:
I think there is just a little too much business logic in this method to make it part of the model object.  prefer to keep it as a helper function on this one.

Reply:
what makes it valid? as in, can you add something descriptive to the name, one more word?

Reply:
there is a few different things going on in there, hard to wrap into one word.  Suggestion?

Reply:
to Thomas' questions, maybe a comment could also suffice.

I'll defer to your judgement in terms of keeping the method here, so no need to move. But two thoughts:
- it seems a lot of this is asking things of the granule record, that could be gotten in the model
- 'they' often say that you should put business logic in models instead of controllers, so I think it would make sense to have business logic there

-

Reply:
I can definitely add a comment.   But what strikes me wrong about moving it to the model is that this is very specific to granules, and the model object is a Record object which could be a granule or collection.  I guess by definition the helper is included in the controller, so could be seen as bloating the controller.   But I consider it more of a helper function to help the application logic determine what to do.

Reply:
Thomas I renamed it to associated_granule_valid? and added comments to all that it is doing.

Reply:
I opted to split this helper function into two different functions and get rid of the flag variable.   I think it makes things a bit more understandable.